datafusion/physical-plan/src/common.rs

```diff
@@ -156,7 +156,11 @@ pub fn compute_record_batch_statistics(
     for partition in batches.iter() {
         for batch in partition {
             for (stat_index, col_index) in projection.iter().enumerate() {
-                null_counts[stat_index] += batch.column(*col_index).null_count();
```

Comment:
The meaning of this line was apparently changed in https://github.com/apache/arrow-rs/pull/4691
At least it seems so from this diff line https://github.com/apache/arrow-rs/commit/979a070dc82eeb26b38a8651cac879b2c276c0ed#diff-0cfddb6ef017ce20c5e7f528095956b2433cf2ea2d17e28ba4b515b7c1bd2e57L179